### ISSUE TITLE
when is "later"?

### ISSUE THREAD

**Issue by nonchip** [YvanTT/arTIfiCE#11]
2 years ago you teased the source code "later", can you tell us when that'll be approximately?

are you waiting for TI to "fix" (=break again) their "vulnerability" (=dropped feature) before making it public or something like that?

**Comment by mpi3d** [comment]
Publishing the source code is definitely not something that takes time. So, I'm only seeing 3 reasons of an unpublished source code:
+ As you mentioned, the code can't be released because it would be too easy for TI to fix it.
+ The source code is trash/draft and need to be reworked
+ In the case of a vulnerability, it might not be a code but more an explanation of the vulnerability. This means that time is needed to write the documentation.

The first one seems to be the main reason. Anyway, @YvanTT has the answer. But he may not reply anytime soon, according to his infrequent activity.

@nonchip You do not seem to work for TI (That's a good sign 🙂), even if we can be too sure. But I'm wondering: Why do you need the source code for?

**Comment by nonchip** [comment]
i don't work for them, am pretty sure of that, and i don't *need* the source code, i'm just curious essentially, and given the "later" was 2 years ago i figured i poke @YvanTT in case they might just have forgotten.

as to your 3 reasons, while i agree those seem the most likely (i can imagine a 4th one: they just don't want anyone to know their secret, but kinda doubt they'd tease it then) in theory, they all seem kinda flawed in practice:

* TI have the source code for Cabri, and also likely have the resources to reverse engineer the hack binary
* there's been 2 years to rework it if it's really that bad
* there's been 2 years to write up the doc, and they called it code, so some source code is probably in there (and even a file format like this could be generated using `.db` statements in any decent assembler)

as to what it actually is, from what i could figure out giving it a quick Ghidra pokearound, it's a Cabri save file that abuses a badly written loader to trick it into jumping to a piece of machine code inside that file.

**Comment by mpi3d** [comment]
Yeah. Anyway, I guess the only thing we can do right now is to wait. So let's wait.

**Comment by nonchip** [comment]
i mean, that's what i've been doing anyway, just felt like replying to your theories is kinda a nice thing to do, after all you took time to come up with them and comment here :P

**Comment by jacks0n9** [comment]
> i don't work for them, am pretty sure of that ;) but also i don't _need_ the source code, i'm just curious essentially, and given the "later" was 2 years ago i figured i poke @YvanTT in case they might just have forgotten.
> 
> as to your 3 reasons, while i agree those seem the most likely (i can imagine a 4th one: they just don't want anyone to know their secret, but kinda doubt they'd tease it then) in theory, they all seem kinda flawed in practice:
> 
> * TI have the source code for Cabri, and also have the resources to reverse engineer the hack binary (and would have had 2 years to do so)
> * there's been 2 years to rework it if it's really that bad
> * there's been 2 years to write up the doc, and they called it code, so some source code is in there (and even a file format like this could be generated using `.db` statements in any decent assembler), but yeah it's probably a combination of "this works because Cabri is broken in such-and-such-way" and "here's the payload.asm"
> 
> as to what it actually is, from what i could figure out giving it a quick Ghidra pokearound, it's a Cabri save file that abuses a badly written loader to trick it into jumping to a piece of machine code inside that file.

yo i tried opening it up in ghidra but i couldn't figure it out (idk how to use ghidra). how did you do that?

**Comment by nonchip** [comment]
@jacks0n9 you'll want to read the tutorial probably, i doubt i can walk you through ghidra in a github issue :D

but yeah after @mpi3d's statements and @YvanTT's silence and given they're in active breach of their own license, i'mma look further into reverse engineering this over in my fork.

**Comment by YvanTT** [comment]
There's no reason for the source code to be public right now, however? I mean, why do you even need it - to build/change what?
If there is a feature that you would need, please open a new issue for this, describing it.

Maybe I mentioned it before, I don't remember, but the plan is to release the source code only once another exploit is needed/published, because it would make this one obsolete/useless. And as long as that is not needed, well...
(This was mentioned [here](https://github.com/YvanTT/arTIfiCE/issues/12#issuecomment-1657922304) correctly)

**Comment by nonchip** [comment]
> There's no reason for the source code to be public right now, however?

https://github.com/YvanTT/arTIfiCE/blob/master/LICENSE

**Comment by YvanTT** [comment]
> > There's no reason for the source code to be public right now, however?
> 
> https://github.com/YvanTT/arTIfiCE/blob/master/LICENSE

As already mentioned (and now further clarified in the README), this license file specifies the license of what's in the git repository files/history, i.e. the website, as anyone can see (so basically, the HTML/CSS/JS files). The downloadable arTIfiCE shell attached in the release has no license specified, and does not mention any open-sourceness either in it (on-calc). The fact that GitHub itself automatically links a "source code" archive (of the repo at the time) is merely coincidental with the release-tracking.  
The GPL license as it is in the repository applies to... the git repository files. Releases (and artifacts/binaries manually uploaded on them) are a GitHub feature built on top of git tags, they are not a git thing. When you fork a repo, you get neither the releases nor their artifacts/binaries, just the tags. Same thing when you clone a repo. And what you get when you clone this repo is what is licensed under the GPL _(to the surprise of nobody)_.

> but yeah after @mpi3d's statements and @YvanTT's silence and given they're in active breach of their own license, i'mma look further into reverse engineering this over in my fork.

This is "great" because in addition to you being wrong _(no, we are not violating the license we chose)_ and doubling-down when you're explained the facts, there is now written evidence that you're actively trying to violate the terms/law/copyright/whatever-you-want-to-call-it.

Let's stop wasting time here. The _actual_ and _original_ source code will be published whenever _we_ feel like it and when we feel it's safe to do so. It will have its own specific license, and will likely even be in its own repo that isn't related to this website one.
